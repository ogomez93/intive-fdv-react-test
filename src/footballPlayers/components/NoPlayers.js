import React from 'react';

const NoPlayers = () => {
  return (
    <tr id="noPlayersMatchFilters">
      <td className="border-2 border-black px-1 py-3 sm:py-6" colSpan="4">No players match those filters</td>
    </tr>
  );
};

export default NoPlayers;