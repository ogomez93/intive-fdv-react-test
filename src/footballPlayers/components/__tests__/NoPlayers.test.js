import React from 'react';
import { shallow } from 'enzyme';
import NoPlayers from '../NoPlayers';

describe('NoPlayers', () => {
  let wrapper;
  
  beforeAll(() => {
    wrapper = shallow(<NoPlayers />);
  });
  
  it('should render', () =>
    expect(wrapper.isEmptyRender()).toEqual(false));

  it('should render correctly', () =>
    expect(wrapper).toMatchSnapshot());

  it('should render a text shows that no players match the filters', () =>
    expect(wrapper.text()).toEqual('No players match those filters'));
});
