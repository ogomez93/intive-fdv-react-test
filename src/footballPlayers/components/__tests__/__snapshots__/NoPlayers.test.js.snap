// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoPlayers should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NoPlayers />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <td
        className="border-2 border-black px-1 py-3 sm:py-6"
        colSpan="4"
>
        No players match those filters
</td>,
      "id": "noPlayersMatchFilters",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "No players match those filters",
        "className": "border-2 border-black px-1 py-3 sm:py-6",
        "colSpan": "4",
      },
      "ref": null,
      "rendered": "No players match those filters",
      "type": "td",
    },
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <td
          className="border-2 border-black px-1 py-3 sm:py-6"
          colSpan="4"
>
          No players match those filters
</td>,
        "id": "noPlayersMatchFilters",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No players match those filters",
          "className": "border-2 border-black px-1 py-3 sm:py-6",
          "colSpan": "4",
        },
        "ref": null,
        "rendered": "No players match those filters",
        "type": "td",
      },
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
